projects/amun/index.js

```diff
@@ -1,41 +1,135 @@
-const sdk = require('@defillama/sdk');
-const abi = require('./abi.json')
+const sdk = require("@defillama/sdk");
+const { default: computeTVL } = require("@defillama/sdk/build/computeTVL");
+const { util } = require("@defillama/sdk");
+const abi = require("./abi.json");
+const axios = require("axios");
+const solana = require("../helper/solana");
+const utils = require("../helper/utils");
+const BigNumber = require("bignumber.js");
 
-const DFI_ADDRESS = '0xA9536B9c75A9E0faE3B56a96AC8EdF76AbC91978';
-const DMX_ADDRESS = '0x1660F10B4D610cF482194356eCe8eFD65B15bA83';
+const PECO_ADDRESS = "0xA9536B9c75A9E0faE3B56a96AC8EdF76AbC91978";
+const DFI_ADDRESS = "0xA9536B9c75A9E0faE3B56a96AC8EdF76AbC91978";
+const SOLI_ADDRESS = "8JnNWJ46yfdq8sKgT1Lk4G7VWkAA8Rhh7LhqgJ6WY41G";
+
+const knownTokenPrices = {};
+
+async function getSolanaTokenRate(address) {
+  return await axios.get(
+    "https://lima-api.amun.com/tokenRate/APygy51vntbQQsYmgkF1CeMsAn5Tv1rsVUB6nee3wJpY?chain=solana&address=" +
+      address
+  );
+}
+
+const locks = [];
+function getCoingeckoLock() {
+  return new Promise((resolve) => {
+    locks.push(resolve);
+  });
+}
+function releaseCoingeckoLock() {
+  const firstLock = locks.shift();
+  if (firstLock !== undefined) {
+    firstLock(null);
+  }
+}
+// Rate limit is 50 calls/min for coingecko's API
+// So we'll release one every 1.2 seconds to match it
+setInterval(() => {
+  releaseCoingeckoLock();
+}, 2000);
+const maxCoingeckoRetries = 5;
 
 function chainTvl(chain, tokens) {
+  if (chain === "solana") {
```

Comment:
I see you've made a hybrid adapter - part SDK and part fetch. Maybe consider using our endpoints for both solana and EVMs - e.g. we have `https://lima-api.amun.com/underlyingTokens/APygy51vntbQQsYmgkF1CeMsAn5Tv1rsVUB6nee3wJpY?chain=solana` endpoint, which can be used to get underlying and their balances for all our tokens. 
We also have `https://lima-api.amun.com/autopilotTokenRate/0xA9536B9c75A9E0faE3B56a96AC8EdF76AbC91978?chain=matic` which returns all the underlyings and their prices, but it currently does not support SOLI

Reply:
we need to deploy this asap, and in a way - the way it is now it will be a good reference on how to do both. 

while, it's not very elegant, as you mentioned, at the moment there is no uniform API to take the relevant data for all the tokens.

once it exists, we may consider to refactor this code (@MiniRoman ?)